chain/car_test.go

```diff
@@ -0,0 +1,190 @@
+package chain_test
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"testing"
+
+	cid "github.com/ipfs/go-cid"
+	ds "github.com/ipfs/go-datastore"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/go-filecoin/address"
+	"github.com/filecoin-project/go-filecoin/chain"
+	tf "github.com/filecoin-project/go-filecoin/testhelpers/testflags"
+	"github.com/filecoin-project/go-filecoin/types"
+)
+
+func TestChainImportExportSimple(t *testing.T) {
+	tf.UnitTest(t)
+	ctx := context.Background()
+
+	cb := chain.NewBuilder(t, address.Undef)
+
+	gene := cb.NewGenesis()
+	ts10 := cb.AppendManyOn(9, gene)
```

Comment:
9 is an arbitrary number here; how is it better than 8?
Cases that are worth testing:
- just the genesis
- genesis plus singleton tip
- genesis plus two singleton tips
- genesis plus a two-block tip
- [gen, 2blocks, 1block]
- [gen, 2block, 2blocks]

Factor out helper methods aggressively so that these cases are described as succinctly as possible.

Some exercising of the messages/nomessages cases as you already have is good too.